File: frontend/heartmage-candles/src/typesV2/order/CandleDetail.ts
import { Candle, Decor, LayerColor, NumberOfLayer, Smell, Wick } from '../BaseProduct';

export interface CandleDetail {
  candle: Candle;
  decor: Decor;
  layerColors?: LayerColor[];
  numberOfLayer: NumberOfLayer;
  smell?: Smell;
  wick: Wick;
}

File: backend/HeartmadeCandles/HeartmadeCandles.Constructor.Core/Models/CandleDetail.cs
namespace HeartmadeCandles.Constructor.Core.Models;

public class CandleDetail
{
    public Candle Candle { get; init; }
    public Decor[] Decors { get; init; }
    public LayerColor[] LayerColors { get; init; }
    public NumberOfLayer[] NumberOfLayers { get; init; }
    public Smell[] Smells { get; init; }
    public Wick[] Wicks { get; init; }
}